Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -404,14 +404,38 @@ void WebCookieJar::setCookieAsync(WebCore::Document& document, const URL& url, c
 }
 
 #if HAVE(COOKIE_CHANGE_LISTENER_API)
-void WebCookieJar::addChangeListener(const String& host, const WebCore::CookieChangeListener& listener)
+bool WebCookieJar::addChangeListenerWithAccess(const URL& url, const URL& firstParty, WebCore::FrameIdentifier frameID, WebCore::PageIdentifier pageID, WebCore::ShouldRelaxThirdPartyCookieBlocking shouldRelaxThirdPartyCookieBlocking, const WebCore::CookieChangeListener& listener)
 {
+    auto host = url.host().toString();
+
+    if (m_changeListeners.contains(host)) {
+        auto listenersForHost = m_changeListeners.get(host);
+        if (listenersForHost.computeSize() > 1 || listenersForHost.contains(listener))
+            return false;
+    }
```

Comment:
Let's use `find()` to reduce one lookup.
`if (auto iter = m_changeListeners.find(host); iter != m_changeListeners.end())`